server-utils/src/cors.rs

```diff
@@ -201,18 +246,99 @@ pub fn get_cors_header(origin: Option<&str>, host: Option<&str>, allowed: &Optio
 						}
 					})
 					.map(|_| AccessControlAllowOrigin::Value(Origin::parse(origin)))
-					.map(CorsHeader::Ok).unwrap_or(CorsHeader::Invalid)
+					.map(AllowOrigin::Ok).unwrap_or(AllowOrigin::Invalid)
+				},
+			}
+		},
+	}
+}
+
+/// Validates if the `AccessControlAllowedHeaders` in the request are allowed.
+pub fn get_cors_allow_headers(request_headers: &Headers, cors_allow_headers: &AccessControlAllowHeaders) -> AllowHeaders {
+	// Check if the header fields which were sent in the request are required
+	if let AccessControlAllowHeaders::Only(only) = cors_allow_headers {
+		let are_all_allowed = request_headers.iter()
+			.all(|header| {
+				only.contains(&Ascii::new(header.name().to_owned())) ||
```

Comment:
Would also be good to avoid copying the header here, Can't we compare `Ascii<String>` with `Ascii<Cow>` or `Ascii<&'a str>`?

Reply:
I changed the type of only to `HashSet<Ascii<&'static str>>`, so it's no longer necessary to copy the header.